run_release.py

```diff
@@ -460,39 +459,62 @@ def create_tag(db: DbfilenameShelf) -> None:
-        the_dir.mkdir(exist_ok=True)
-        filename = f"Python-{db['release']}"
-        tarball = f"Python-{db['release']}.tgz"
-        shutil.copy2(
-            db["git_repo"] / str(db["release"]) / "src" / tarball,
-            the_dir / tarball,
-        )
-        subprocess.check_call(["tar", "xvf", tarball], cwd=the_dir)
-        subprocess.check_call(
-            ["./configure", "--prefix", str(the_dir / "installation")],
-            cwd=the_dir / filename,
-        )
-        subprocess.check_call(["make", "-j"], cwd=the_dir / filename)
-        subprocess.check_call(["make", "install", "-j"], cwd=the_dir / filename)
-        process = subprocess.run(
-            ["./bin/python3", "-m", "test", "-uall"],
-            cwd=str(the_dir / "installation"),
-            text=True,
-        )
+def wait_for_source_and_docs_artifacts(db: DbfilenameShelf) -> None:
+    # Determine if we need to wait for docs or only source artifacts.
+    release_tag = db["release"]
+    should_wait_for_docs = release_tag.is_final or release_tag.is_release_candiate
 
-    if process.returncode == 0:
-        return
+    # Create the directory so it's easier to place the artifacts there.
+    release_path = pathlib.Path(db["git_repo"] / str(db["release"]))
+    release_path.mkdir(parents=True, exist_ok=True)
 
-    if not ask_question("Some test_failed! Do you want to continue?"):
-        raise ReleaseException("Test failed!")
+    # Build the list of filepaths we're expecting.
+    wait_for_paths = [
+        release_path / "src" / f"Python-{db['release']}.tgz",
+        release_path / "src" / f"Python-{db['release']}.tar.xz"
+    ]
+    if should_wait_for_docs:
+        wait_for_paths.extend([
+            release_path / "docs" / f"python-{db['release']}-docs.epub",
+            release_path / "docs" / f"python-{db['release']}-docs-html.tar.bz2",
+            release_path / "docs" / f"python-{db['release']}-docs-html.zip",
+            release_path / "docs" / f"python-{db['release']}-docs-pdf-a4.tar.bz2",
+            release_path / "docs" / f"python-{db['release']}-docs-pdf-a4.zip",
+            release_path / "docs" / f"python-{db['release']}-docs-pdf-letter.tar.bz2",
+            release_path / "docs" / f"python-{db['release']}-docs-pdf-letter.zip",
+            release_path / "docs" / f"python-{db['release']}-docs-texinfo.tar.bz2",
+            release_path / "docs" / f"python-{db['release']}-docs-texinfo.zip",
+            release_path / "docs" / f"python-{db['release']}-docs-text.tar.bz2",
+            release_path / "docs" / f"python-{db['release']}-docs-text.zip",
+        ])
+
+    print(f"Waiting for source{' and docs' if should_wait_for_docs else ''} artifacts to be built")
+    print(f"Artifacts should be placed at '{release_path}':")
+    for path in wait_for_paths:
+        print(f"- '{os.path.relpath(path, release_path)}'")
+
+    while not all(path.exists() for path in wait_for_paths):
+        time.sleep(1)
```

Comment:
Maybe this should exit if after a certain amount of seconds some files are still missing (e.g. because they haven't been created due to some error)?

Reply:
This follows the same model as the "waiting for binary installers" routine which waited forever. I'm not sure how release managers would like to use this tool so would need feedback if there's a preference.